src/http/http_parser.h

```diff
@@ -77,6 +77,11 @@ namespace http
     s.resize(dst - s.data());
   }
 
+  static bool status_ok(http_status status)
+  {
+    return status >= 200 && status < 300;
+  }
```

Comment:
not sure this should be in this file and not sure it should be called `status_ok`, @eddyashton ?

Reply:
This file's fine, but I think `status_success` rather than `status_ok` (`OK` is a single status, 200, and the general term for the `2xx` family seems to be 'Success').